### PR TITLE
feat(cloudtrail): enable CloudTrail Insights on Trail

### LINKED ISSUE #8335: Enable CloudTrail Insights on Trail
Enable CloudTrail Insights when creating a Trail. 

### Proposed Solution

Could be a property of TrailProps.

```
const trail = new cloudtrail.Trail(this, 'CloudTrail', {
  enableInsights: true
});
```

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by watany-dev**
After an issue is submitted, the specification is added and the implementation is such that the InsightType is selected.

fixes #8335 

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by Naumel** [comment]
Mergifyio update

**Comment by Naumel** [comment]
I believe the following
```
aws-cdk-lib: aws-cloudtrail/test/integ.cloudtrail-insight.lit.ts:4:24 - error TS2307: Cannot find module '../../integ-tests' or its corresponding type declarations.
aws-cdk-lib: 4 import * as integ from '../../integ-tests';
aws-cdk-lib:                          ~~~~~~~~~~~~~~~~~~~
```
is relevant to the changes here.

**Comment by watany-dev** [comment]
Hmmm, this error may not be immediately obvious because the integtest I have modified is not a relative path specification as shown below.
aws-cdk-lib:4 import * as integ from '... /... /integ-tests';

Suspicious part is aws-cdk-lib instead of @aws-cdk-lib?

**Comment by mrgrain** [comment]
> Hmmm, this error may not be immediately obvious because the integtest I have modified is not a relative path specification as shown below. aws-cdk-lib:4 import * as integ from '... /... /integ-tests';
> 
> Suspicious part is aws-cdk-lib instead of @aws-cdk-lib?

Integration tests with the `.lit.ts` extensions are an older style that is also used in documentation examples. You'll have to do either one of:

- Find the examples by searching for the file name (without ending) and replace them with hand-crafted inline examples
- Remove the `.lit` from the filename

or

- Write a completely new integration test without the `.lit` thingy

**Comment by watany-dev** [comment]
I am very pleased with your kind response. I will try the former method to resolve this.

File: packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts [hunk 1]
```diff
@@ -158,6 +165,23 @@ export enum ReadWriteType {
   NONE = 'None',
 }
 
+/**
+ * Util element for InsightSelector
+ */
+export class InsightType {
```
[the remark this answers is not in the document]
Comment (watany-dev) [reply packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts:171]: My apologies. I'm just left with something that was created with expandability in mind during development.

**Pull request merged**